apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/DbTransactionHelper.java

```diff
@@ -30,6 +30,19 @@ public static void executeInTransaction(TransactionalDataSource dataSource, Tran
             throw new DbTransactionExecutionException(e.toString(), e);
         }
     }
+    public static<T> T executeInConnection(TransactionalDataSource dataSource, TransactionFunction<T> op) {
+        TransactionalDataSource.StartedConnection startedConnection = dataSource.beginTransactionIfNotStarted();
+        try {
+            T result = op.execute();
+            if (!startedConnection.isAlreadyStarted()) {
+                dataSource.commit(true);
```

Comment:
I'm not sure for 100% that we have to close connection here after commit. May be better to discuss in team.

Reply:
It's okay for cases when an existing transaction should be maintained